lutris/services/rom.py

```diff
@@ -0,0 +1,106 @@
+from mmap import mmap
+import os
+from lutris import pga
+from lutris.util.log import logger
+from lutris.util.strings import slugify
+from lutris.config import make_game_config_id, LutrisConfig
+
+NAME = "rom"
+INSTALLER_SLUG = "rom"
+TDB_DB_CACHE = None
+
+def add_rom(rom, config):
+    """ add a dolhin rom to the database ( update if game already exist )
+    rom and config are list, respectively for the rom data and the runner config """
+    logger.info("adding %s."%str(rom["name"]))
+
+    config_id = make_game_config_id(rom["slug"])
+    rom["configpath"] = config_id
+
+    pga.add_or_update(**rom)
+
+    game_config = LutrisConfig(
+        runner_slug=rom["runner"],
+        game_config_id=config_id
+    )
+    game_config.raw_game_config.update(config)
+    game_config.save()
+
+
+def rom_read_data(location):
+    """ extract data from the rom location at location.
+    return a dict with "data" and "config", to be applied to a game in Lutris """
+    # TODO: extract the image of the rom
+    def scan_to_00(mm, start):
```

Comment:
What is the reasoning behind having these two functions nested in rom_read_data?

Reply:
I've put this here because that made them private ( but maybe put them globally could be useful for other use ).